frontend/src/components/header/Header.tsx

```diff
@@ -42,9 +42,9 @@ const Header: React.FC<HeaderProps> = ({isLogged}: HeaderProps) => {
 
     const redirectToPage = (isLogged: boolean, navigate: NavigateFunction) => {
         if (isLogged) {
-            navigate(Paths.Home);
-        } else if (location.pathname !== Paths.SignUp) {
-            navigate(Paths.SignIn);
+            navigate(Paths.HOME);
```

Comment:
It's totally not user friendly. If user is on profile page and decided to reload the page you route them to home page. But they want to stay on the same page where they were.

If use logged in, check if path is not within your main paths and move them to home page. Otherwise, you don't need to route them anywhere. Same for auth.